### PR TITLE
Fix crash issue for bnns

### PR DESCRIPTION
**Pull request by lisa0314**
Fix [intel/webml-polyfill#550](https://github.com/intel/webml-polyfill/issues/550).

- ResNet50 V1(onnx) worked.
- ResNet50 V2(onnx) worked.
- Inception Resnet v2(TFlite) worked.

@fujunwei , please help to review.

### CODE REVIEW

**Comment by fujunwei** [comment]
Could you please explain what's the reason of the crashed issue.

**Comment by lisa0314** [comment]
@fujunwei it's crashed at line 103. Since the inputs of the operand of add operation in models are the output of the other layer, it can't be got by `memory get`.

File: services/ml/compilation_impl_mac_bnns.mm [hunk 1]
```diff
@@ -97,10 +98,12 @@ bool CompileCompileArithmeticBNNS(OperationMac& operation,
...
-  uint32_t extend_input_idx =
-      values.find(inputs[0]) == values.end() ? inputs[1] : inputs[0];
-  operation.extend_input.push_back(reinterpret_cast<float*>(
-      memory.get() + values.at(extend_input_idx).offset));
+  if (is_first_operation) {
+    uint32_t extend_input_idx =
+        values.find(inputs[0]) == values.end() ? inputs[1] : inputs[0];
+    operation.extend_input.push_back(reinterpret_cast<float*>(
+        memory.get() + values.at(extend_input_idx).offset));
+  }
```
Comment (fujunwei) [inline services/ml/compilation_impl_mac_bnns.mm]: I think it's better to use 

```
 for (size_t i = 0; i < 2; ++i) {
    size_t extend_input_idx= inputs[i];
    if (values.find(extend_input_idx) != values.end()) {
      operation.extend_input.push_back(reinterpret_cast<float*>(
            memory.get() + values.at(extend_input_idx).offset));
    }
}
```
and remove  `is_first_operation`

Comment (lisa0314) [reply services/ml/compilation_impl_mac_bnns.mm]: But this solution will lead to more time consuming.

File: services/ml/compilation_impl_mac_bnns.mm [hunk 2]
```diff
@@ -91,16 +92,20 @@ bool CompileCompileArithmeticBNNS(OperationMac& operation,
...
+      operation.extend_input.push_back(reinterpret_cast<float*>(
+          memory.get() + values.at(extend_input_idx).offset));
+    }
   }
 
-  uint32_t extend_input_idx =
-      values.find(inputs[0]) == values.end() ? inputs[1] : inputs[0];
-  operation.extend_input.push_back(reinterpret_cast<float*>(
-      memory.get() + values.at(extend_input_idx).offset));
+  if (is_first_operation) {
```
Comment (fujunwei) [inline services/ml/compilation_impl_mac_bnns.mm]: Remove the case codes that is refactored with l.95.

**Comment by lisa0314** [comment]
@fujunwei , please help to review again. Thanks!

**Comment by fujunwei** [comment]
LGTM, thanks, merging.

**Pull request merged**